orderBook.js

```diff
@@ -0,0 +1,47 @@
+/* eslint-disable max-len */
+function reconcileOrder(existingBook, incomingOrder) {
+  let updatedBook = []
+
+  if (existingBook.length <= 0) {
+    return existingBook.concat(incomingOrder)
+  }
+
+  for (let i = 0; i < existingBook.length; i++) {
+    if (incomingOrder.type === existingBook[i].type) {
+      existingBook.push(incomingOrder)
+
+      return existingBook
+    }
+
+    if ((incomingOrder.type !== existingBook[i].type) && (incomingOrder.price !== existingBook[i].price)) {
```

Comment:
(incomingOrder.type !== existingBook[i].type) is repeated many times..

can move it to one parent if statement (still inside for loop) and all other checks within it.

example: 

if (incomingOrder.type !== existingBook[i].type){
   if (incomingOrder.price !== existingBook[i].price){
   ....
   }else if((incomingOrder.price === existingBook[i].price) && (incomingOrder.quantity === existingBook[i].quantity){
   ...
   }
   
}